internal/controller/numaflowcontroller/numaflowcontroller_controller_test.go

```diff
@@ -16,4 +16,376 @@ limitations under the License.
+
+	restConfig, numaflowClientSet, client, k8sClientSet, err := commontest.PrepareK8SEnvironment()
+	assert.Nil(t, err)
+	assert.Nil(t, kubernetes.SetDynamicClient(restConfig))
+
+	getwd, err := os.Getwd()
+	assert.Nil(t, err, "Failed to get working directory")
+	configPath := filepath.Join(getwd, "../../../", "tests", "config")
+	configManager := config.GetConfigManagerInstance()
+	err = configManager.LoadAllConfigs(func(err error) {}, config.WithConfigsPath(configPath), config.WithConfigFileName("testconfig"))
+	assert.NoError(t, err)
+
+	controllerDefinitions, err := ctlrcommon.GetNumaflowControllerDefinitions("../../../tests/config/controller-definitions-config.yaml")
+	assert.Nil(t, err)
+	config.GetConfigManagerInstance().GetControllerDefinitionsMgr().UpdateNumaflowControllerDefinitionConfig(*controllerDefinitions)
+
+	ctx := context.Background()
+
+	// other tests may call this, but it fails if called more than once
+	if ctlrcommon.TestCustomMetrics == nil {
+		ctlrcommon.TestCustomMetrics = metrics.RegisterCustomMetrics()
+	}
+
+	recorder := record.NewFakeRecorder(64)
+
+	r, err := NewNumaflowControllerReconciler(
+		client,
+		scheme.Scheme,
+		restConfig,
+		kubernetes.NewKubectl(),
+		ctlrcommon.TestCustomMetrics,
+		recorder,
+	)
+	assert.NoError(t, err)
+
+	trueValue := true
+
+	pipelinerollout.PipelineROReconciler = &pipelinerollout.PipelineRolloutReconciler{Queue: util.NewWorkQueue("fake_queue")}
+
+	testCases := []struct {
+		name                    string
+		newControllerVersion    string             // new version specified in NumaflowController
+		existingController      *appsv1.Deployment // if nil, then there isn't a Controller installed
+		existingPipelineRollout *apiv1.PipelineRollout
+		existingPipeline        *numaflowv1.Pipeline
+		existingPauseRequest    *bool // was Numaflow Controller previously requesting pause?
+		expectedPauseRequest    *bool // after reconcile(), should it be requesting pause?
+		expectedPhase           apiv1.Phase
+		// require these Conditions to be set (note that in real life, previous reconciliations may have set other Conditions from before which are still present)
+		expectedConditionsSet           map[apiv1.ConditionType]metav1.ConditionStatus
+		expectedResultControllerVersion string // the one that's been deployed
+		expectedReconcileError          bool
+	}{
+		{
+			name:                    "no existing Controller",
+			newControllerVersion:    "1.2.0",
+			existingController:      nil,
+			existingPipelineRollout: nil,
+			existingPipeline:        nil,
+			existingPauseRequest:    nil,
+			expectedPauseRequest:    nil,
+			expectedPhase:           apiv1.PhaseDeployed,
+			expectedConditionsSet: map[apiv1.ConditionType]metav1.ConditionStatus{
+				apiv1.ConditionChildResourceDeployed: metav1.ConditionTrue,
+			},
+			expectedResultControllerVersion: "1.2.0",
+		},
+		{
+			name:                 "new Controller version, pipelines paused",
```

Comment:
I see the unit test involves pausing - was this brought over from before or did you need to modify it? I'm wondering if it should still go in NumaflowControllerRollout's unit test, no?

Reply:
If we simply removed this function, I'm guessing we would probably still have the same code coverage due to the other test?

Reply:
Sure, I'll remove those unit tests from this file then.